haystack/preview/components/preprocessors/text_document_splitter.py

```diff
@@ -0,0 +1,109 @@
+from typing import List, Optional, Dict, Any, Literal, Tuple
+
+from more_itertools import windowed
+
+from haystack.preview import component, Document, default_from_dict, default_to_dict
+
+
+@component
+class TextDocumentSplitter:
+    """
+    Split a text document into a list of text documents with shorter texts.
+    This is useful for splitting documents with long texts that otherwise would not fit into the maximum text length of language models
+    """
+
+    def __init__(
+        self, split_by: Literal["word", "sentence", "passage"] = "word", split_length: int = 200, split_overlap: int = 0
+    ):
+        """
+        :param split_by: The unit by which the document should be split. Choose from "word", "sentence", "passage".
```

Comment:
Let's clarify the meaning of these three values and their limitations, i.e.: `word` implies splitting by whitespace, `sentence` splitting by dot and `passage`/`paragraph` splits by double-return